### PR TITLE
Personal details refactor

### PR DESCRIPTION
**Pull request by Lansalo**
Added some extra unit test coverage for personal details repository.
Also rename methods to make explicit what they actually do.
A possible improvement would be to have just one "update" method in the repository and have the service to do the status update.

### CODE REVIEW

File: app/repositories/application/PersonalDetailsRepository.scala [hunk 1]
```diff
@@ -74,27 +74,10 @@ class PersonalDetailsMongoRepository(implicit mongo: () => DB)
...
-
-  def update(applicationId: String, userId: String, personalDetails: PersonalDetails,
-    requiredApplicationStatuses: Seq[ApplicationStatuses.EnumVal],
-    newApplicationStatus: ApplicationStatuses.EnumVal
-  ): Future[Unit] = {
+  def updatePersonalDetailsAndStatus(applicationId: String, userId: String, personalDetails: PersonalDetails,
+             requiredApplicationStatuses: Seq[ApplicationStatuses.EnumVal],
+             newApplicationStatus: ApplicationStatuses.EnumVal
+            ): Future[Unit] = {
     val PersonalDetailsCollection = "personal-details"
```
Comment (mgilgar) [inline app/repositories/application/PersonalDetailsRepository.scala]: You can have this val as a class field and reuse it in the rest of the methods, for instance in line 111

File: it/repositories/application/PersonalDetailsMongoRepositorySpec.scala [hunk 2]
```diff
@@ -20,22 +23,60 @@ class PersonalDetailsMongoRepositorySpec extends MongoRepositorySpec with UUIDFa
...
-        aLevel = false, stemLevel = false, civilServant = false, department = None)
-      repository.update(appId, userId, personalDetails).futureValue
+      repository.updatePersonalDetailsAndStatus(appId, userId, ExpectedPersonalDetails).futureValue
 
       val result = repository.find(appId).futureValue
-      result mustBe personalDetails
+      result mustBe ExpectedPersonalDetails
     }
 
     "find a candidate after he has been updated (civil servant)" in {
```
Comment (mgilgar) [inline it/repositories/application/PersonalDetailsMongoRepositorySpec.scala]: I think this test looks now the same as the one in line 21: "find a candidate after he has been updated (not civil servant)". Probably you can remove one.

File: it/repositories/application/PersonalDetailsMongoRepositorySpec.scala [hunk 3]
```diff
@@ -20,22 +23,60 @@ class PersonalDetailsMongoRepositorySpec extends MongoRepositorySpec with UUIDFa
...
-      repository.update(appId, userId, personalDetails).futureValue
+
+      repository.updatePersonalDetailsAndStatus(appId, userId, ExpectedPersonalDetails).futureValue
 
       val result = repository.find(appId).futureValue
-      result mustBe personalDetails
+      result mustBe ExpectedPersonalDetails
+    }
+
+    "update personal details and status" in {
```
Comment (mgilgar) [inline it/repositories/application/PersonalDetailsMongoRepositorySpec.scala]: I think the next 2 tests (line 41 and 51) probably supersede the previous two (line 21 and line 31)

File: app/repositories/application/PersonalDetailsRepository.scala (hunk 1 above)
Comment (gnebro) [inline app/repositories/application/PersonalDetailsRepository.scala]: Replaced with the literal instead as it's easy to read and it matches the convention we follow here and in other places when we write BSONDocument content.

File: it/repositories/application/PersonalDetailsMongoRepositorySpec.scala (hunk 2 above)
Comment (gnebro) [inline it/repositories/application/PersonalDetailsMongoRepositorySpec.scala]: Done

File: it/repositories/application/PersonalDetailsMongoRepositorySpec.scala (hunk 3 above)
Comment (Lansalo) [inline it/repositories/application/PersonalDetailsMongoRepositorySpec.scala]: Done

**Comment by Lansalo** [comment]
@mgilgar Ok, I first run the acceptance criteria against the most updated fersion from masted of backend and admin frontend and at then with this PR. In both cases no failures.

**Pull request merged**